onprc_ehr/src/org/labkey/onprc_ehr/notification/ColonyAlertsNotification.java

```diff
@@ -1219,6 +1219,122 @@ public void exec(ResultSet object) throws SQLException {
+        if (count > 0) {
+            Set<FieldKey> columns = new HashSet<>();
+            columns.add(FieldKey.fromString("Id"));
+            columns.add(FieldKey.fromString("Indate"));
+            columns.add(FieldKey.fromString("Building"));
+            columns.add(FieldKey.fromString("Area"));
+            columns.add(FieldKey.fromString("Room"));
+            columns.add(FieldKey.fromString("Cage"));
+            columns.add(FieldKey.fromString("housingType"));
+            columns.add(FieldKey.fromString("housingCondition"));
+            columns.add(FieldKey.fromString("ReasonForMove"));
+            columns.add(FieldKey.fromString("Remark"));
+            columns.add(FieldKey.fromString("TotalAnimals"));
+            columns.add(FieldKey.fromString("ActiveClinicalTreatment"));
+
+            final Map<FieldKey, ColumnInfo> colMap = QueryService.get().getColumns(ti, columns);
+            TableSelector ts2 = new TableSelector(ti, colMap.values(), null, null);
+            count = ts2.getRowCount();
+
+            msg.append("<br><b>Housing Transfers:</b><br><br>");
+            //Legend
+            msg.append("<table border=1 style='border-collapse: collapse;'>");
+            msg.append("<tr bgcolor = " + '"' + "#FFFF00" + '"' + ">");
+            msg.append("<td> Yellow indicates animal transferred into an empty room</td></tr>");
+            msg.append("<tr bgcolor = " + '"' + "#00FFFF" + '"' + ">");
+            msg.append("<td> Blue indicates animal has active clinical treatments</td></tr>");
+            msg.append("</table>");
+            //Header
+            msg.append("<br><br><table border=1 style='border-collapse: collapse;'>");
+            msg.append("<tr bgcolor = " + '"' + "#00FF7F" + '"' + "style='font-weight: bold;'>");
+            msg.append("<td> Id </td><td> In Date </td><td> Building </td><td> Area </td><td> Room </td><td> Cage </td><td> Housing Type </td><td> Housing Condition </td><td> Reason For Move </td><td> Remark </td><td> Num animals before the transfer </td></tr>");
+
+            ts2.forEach(new Selector.ForEachBlock<ResultSet>() {
+                @Override
+                public void exec(ResultSet object) throws SQLException {
+                    Results rs = new ResultsImpl(object, colMap);
+                    String url = getParticipantURL(c, rs.getString("Id"));
+                    int animal_count = Integer.parseInt(rs.getString("TotalAnimals"));
+
+                    if ( animal_count == 0) { //high light the row in yellow if the room was empty before the move
+                        msg.append("<tr bgcolor = " + '"' + "#FFFF00" + '"' + ">");
+                        if (rs.getString("ActiveClinicalTreatment") != null) { //High light with blue color if there are any active treatments
+                            msg.append("<td style= " + '"' + "background-color:#00FFFF" +'"' + "> <a href='" + url + "'>" + rs.getString("Id") + "</a></td>\n");
```

Comment:
All of these calls to rs.getString() should HTML encode the values that come back. Most of the time it won't make any difference because the values won't include special HTML characters, but they could.
```suggestion
                            msg.append("<td style= " + '"' + "background-color:#00FFFF" +'"' + "> <a href='" + url + "'>" + PageFlowUtil.filter(rs.getString("Id")) + "</a></td>\n");
```